test/algorithms/length/length_sph.cpp

```diff
@@ -0,0 +1,85 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+// Unit Test
+
+// Copyright (c) 2016 Oracle and/or its affiliates.
+// Contributed and/or modified by Vissarion Fisikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#include <algorithms/test_length.hpp>
+
+#include <boost/geometry/geometries/geometries.hpp>
+#include <boost/geometry/geometries/point_xy.hpp>
+#include <boost/geometry/geometries/adapted/std_pair_as_segment.hpp>
+
+#include <test_geometries/all_custom_linestring.hpp>
+#include <test_geometries/wrapped_boost_array.hpp>
+
+std::vector<std::string > Ls_data_gen()
+{
+   std::string arr[] = {"LINESTRING(0 0,180 0,180 180)",
+                        "LINESTRING(0 0,180 0,180 0,180 0,180 180,180 180)",
+                        "LINESTRING(0 0,180 0,180 10,180 20,180 30,180 40,180 50,180 60,180 70,180 80,180 90,180 100,\
+                        180 110,180 120,180 130,180 140,180 150,180 160,180 170,180 180)"};
+   std::vector<std::string> Ls_data (arr, arr + sizeof(arr) / sizeof(arr[0]));
+   return Ls_data;
+}
+
+template <typename P>
+void test_all_default() //test the default strategy
+{
+    double const pi = boost::math::constants::pi<double>();
+    std::vector<std::string > Ls_data = Ls_data_gen();
+
+    for(size_t i=0; i<2; ++i)
```

Comment:
Replace `size_t` with `std::size_t`.
Please add spaces before/after `=` and `<`.